tests/connection.rs

```diff
@@ -202,3 +202,65 @@ fn server_inactive() {
 
     assert_eq!(app.world().resource::<ServerTick>().get(), 0);
 }
+
+#[test]
+fn after_deferred_replication_start() {
+    let mut server_app = App::new();
+    let mut client_app = App::new();
+    for app in [&mut server_app, &mut client_app] {
+        app.add_plugins((
+            MinimalPlugins,
+            RepliconPlugins.set(ServerPlugin {
+                tick_policy: TickPolicy::EveryFrame,
+                replicate_after_connect: false,
+                ..Default::default()
+            }),
+        ));
+    }
+
+    server_app.connect_client(&mut client_app);
+
+    let client_id = client_app
+        .world()
+        .resource::<RepliconClient>()
+        .id()
+        .unwrap();
+
+    server_app.update();
+    server_app.exchange_with_client(&mut client_app);
+    client_app.update();
+    server_app.exchange_with_client(&mut client_app);
+
+    // Test that we're not replicating to this client yet.
```

Comment:
I suggest to put comment like this into the assert message.
But it will need to be rephrased a little like this: "we shouldn't replicate to this client yet" because panic messages in Rust usually follow this pattern.